### ISSUE TITLE
args[module] is confused when there is a module and an action plugin

### ISSUE THREAD

**Issue by evgeni** [ansible/ansible-lint#2824]
##### Summary

When there is an action plugin and a module by the same name, Ansible calls the action plugin which then will call the module internally.

These two can, however, have different parameters, like we see in the `async_status` combo. The action plugin has two: `jid` and `mode`, while the module has a third, hidden one: `_async_dir`.

##### Issue Type

- Bug Report

##### OS / ENVIRONMENT

```console 
% ansible-lint --version     
ansible-lint 6.10.0 using ansible 2.14.1
```

- ansible installation method: pip
- ansible-lint installation method: pip

##### STEPS TO REPRODUCE

This is the example from `ansible-doc async_status`:

```yaml
---
- name: Lol
  hosts: localhost
  tasks:
    - name: Asynchronous yum task
      ansible.builtin.yum:
        name: docker-io
        state: present
      async: 1000
      poll: 0
      register: yum_sleeper

    - name: Wait for asynchronous job to end
      ansible.builtin.async_status:
        jid: '{{ yum_sleeper.ansible_job_id }}'
      register: job_result
      until: job_result.finished
      retries: 100
      delay: 10
```

```console
% ansible-lint l2.yml   
WARNING  Overriding detected file kind 'yaml' with 'playbook' for given positional argument: l2.yml
WARNING  Listing 1 violation(s) that are fatal
args[module]: missing required arguments: _async_dir (warning)
l2.yml:13 Task/Handler: Wait for asynchronous job to end

You can skip specific rules or tags by adding them to your configuration file:
# .config/ansible-lint.yml
warn_list:  # or 'skip_list' to silence them completely
  - experimental  # all rules tagged as experimental

                  Rule Violation Summary                   
 count tag          profile rule associated tags           
     1 args[module]         syntax, experimental (warning) 

Passed with production profile, 5/5 star rating: 0 failure(s), 1 warning(s) on 1 files.
```

##### Desired Behavior

`args[module]` should not be raised

##### Actual Behavior

```
args[module]: missing required arguments: _async_dir (warning)
```

**Comment by evgeni** [comment]
A similar issue happens with `ansible.builtin.service`.
The doc of `ansible.builtin.service` says: "This module acts as a proxy to the underlying service manager module. While all arguments will be passed to the underlying module, not all modules support the same arguments."

The following invocation works just fine (`daemon_reload` is accepted and passed to `systemd`):
```yaml
- name: Restart rhcd
  ansible.builtin.service:
    name: rhcd
    state: restarted
    daemon_reload: true
```

But `ansible-lint` complains "Unsupported parameters for ansible.builtin.service module: daemon_reload. Supported parameters include: sleep, runlevel, enabled, name, pattern, state, arguments (args)." which is true

**Comment by goldyfruit** [comment]
Same here.

```yaml
    - name: Start QEMU instance
      ansible.builtin.shell:
        cmd: qemu-system-aarch64
          -M raspi3b
          -m 1024
          -kernel /tmp/{{ qemu_kernel }}
          -dtb /tmp/{{ qemu_dtb }}
          -sd ../recovery-rpi4b.img
          -append "console=ttyAMA0 root=/dev/mmcblk0p2 rw rootwait rootfstype=ext4"
          -nographic
          -device usb-net,netdev=net0
          -netdev user,id=net0,hostfwd=tcp::{{ qemu_port }}-:22
        executable: /bin/bash
      register: _qemu_cmd
      changed_when: true
      async: 1
      poll: 0

    - name: Retrieve "Start QEMU instance" task status
      ansible.builtin.async_status:
        jid: "{{ _qemu_cmd.ansible_job_id }}"
      register: _qemu_task_status
      failed_when: _qemu_task_status.rc > 0
```

`ansible-lint` output:

```
WARNING  Overriding detected file kind 'yaml' with 'playbook' for given positional argument: ansible/recovery-post.yml
WARNING  Listing 1 violation(s) that are fatal
args[module]: missing required arguments: _async_dir (warning)
ansible/recovery-post.yml:31 Task/Handler: Retrieve "Start QEMU instance" task status

You can skip specific rules or tags by adding them to your configuration file:
# .config/ansible-lint.yml
warn_list:  # or 'skip_list' to silence them completely
  - experimental  # all rules tagged as experimental

                  Rule Violation Summary                   
 count tag          profile rule associated tags           
     1 args[module]         syntax, experimental (warning) 

Passed with production profile, 5/5 star rating: 0 failure(s), 1 warning(s) on 1 files.
A new release of ansible-lint is available: 6.10.0 → 6.10.1
```

**Comment by cidrblock** [comment]
How about this as an approach?

1) Continue to check the task's arg
2) If there is a failure, and an action with the same name, only log that there was a failure but an error was not raised

The low-level issue here is that there's no way to know what the behavior of the action is without running it. I think the best we can do is let the user know there _might_ be an issue, but we don't know for sure.

**Comment by ssbarnea** [comment]
Closing as not being able to reproduce with current version.
